nautobot/core/templates/generic/object_list.html

```diff
@@ -163,7 +195,38 @@ <h1>{% block title %}{{ content_type.model_class|meta:"verbose_name_plural"|bett
         // By removing all name attr with 'form-' prefix form input/select elements
         // only filterable fields would be picked up by `$(this).serialize()`
         $(this).find(`input[name*="form-"], select[name*="form-"]`).removeAttr("name")
-        location.replace("?" + $(this).serialize())
+
+        let query_params = $(this).serialize()
+
+        // check if `?q=` or `&q=` in query_params
+        if(location.search && /(\?|\&)q/.test(location.search)){
+            // Append `q` value to `query_params`
+            query_params += "&q=" + /(?<=(\?|\&)q=)\w+/.exec(location.search)[0]
+        }
+        location.replace("?" + query_params)
+    })
+
+    // On submit of filter search form
+    $("#search-form").on("submit", function(e){
+        e.preventDefault();
+
+        let query_params = location.search;
+        let search_value = $(this).find("input[name='q']").val();
+
+        if(query_params){
+            // check if `?q=` or `&q=` in query_params
+            if(/(\?|\&)q/.test(query_params)){
+                // change value of q in the query string
+                query_params = query_params.replace(/(?<=(\?|\&)q=)\w+/, search_value);
+            }
+            else {
+                query_params += "&q=" + search_value;
```

Comment:
Couldn't this be, for more self-consistency:

```suggestion
                query_params += "&" + $(this).serialize();
```